engine/src/main/java/org/terasology/engine/world/block/BlockArea.java

```diff
@@ -354,4 +355,35 @@ public int hashCode() {
     public String toString() {
         return "BlockArea[(" + this.minX + ", " + this.minY + ")...(" + this.maxX + ", " + this.maxY + ")]";
     }
+
+    private int calDis(int x, int y){
+        return (int) Math.sqrt(x * x + y * y);
+    }
+    public int distanceSquared(int px, int py){
```

Comment:
I would also add some unit test to validate if this is working as intended.

Reply:
Agreed, see https://github.com/MovingBlocks/Terasology/pull/4621#discussion_r622032468 above.